### PR TITLE
Convert button `.icon` and `.tonalIcon` constructors to take nullable icons.

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

This changes the factory constructors for `TextButton.icon`, `ElevatedButton.icon`, `FilledButton.icon`, and `FilledButton.tonalIcon` to take nullable icons. If the icon is null, then the "regular" version of the button is created.

## Tests
 - Added tests for all four constructors.

### CODE REVIEW

**Comment by HansMuller** [comment]
Would making the TextButton.icon `icon` parameter nullable work?

**Comment by gspencergoog** [comment]
> Would making the TextButton.icon `icon` parameter nullable work?

I could probably do that. I'll see if I can.  I was thinking that the `TextButton.icon` constructor was const, but I guess it isn't.

File: packages/flutter/lib/src/material/filled_button.dart [hunk 1]
```diff
@@ -92,9 +94,38 @@ class FilledButton extends ButtonStyleButton {
...
+        onFocusChange: onFocusChange,
+        style: style,
+        focusNode: focusNode,
+        autofocus: autofocus ?? false,
+        clipBehavior: clipBehavior ?? Clip.none,
+        statesController: statesController,
+        child: label,
+      );
+    }
+    return _FilledButtonWithIcon( key: key,
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/filled_button.dart:115]: Maybe put `key: key` on the next line like the previous constructor call?

**Review by HansMuller — APPROVED** [review]
This LGTM although making the same changes in ElevatedButton.icon and OutlinedButton.icon seems warranted.

File: packages/flutter/lib/src/material/filled_button.dart (hunk 1 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/material/filled_button.dart:115]: Whoops, just an oversight.

**Pull request merged**